### ISSUE TITLE
Nextcloud want's to download my file instead of open it

### ISSUE THREAD

**Issue by EgoLeX** [jhass/nextcloud-keeweb#116]
I got the above discribed problem. My *.kdbx file is waiting in Nextcloud to be opened but Nextcloud only want to open it.

Some ideas?

**Comment by ithurtswhen-IP** [comment]
Me2.
Nextcloud: 16.0.5
[mimetype-detection]https://github.com/jhass/nextcloud-keeweb#mimetype-detection is set.

I do not have ssh access to the server.
Is `sudo -u www-data php occ files:scan --all` mandatory for this function?

**Comment by nub3** [comment]
I, too,  ran into the very same issue with Nextcloud 17. However the solution was quite simple, as stated here: https://docs.nextcloud.com/server/17/admin_manual/configuration_mimetypes/index.html

When using docker based Nextcloud - like in my case - simply enter
`docker exec --user www-data -it $CONTAINER_ID php occ maintenance:mimetype:update-js`

To obtain the $CONTAINER_ID, just do a `docker ps | grep nextcloud_app`. The first listed hex number is the docker container id of the running Nextcloud intance.

Hope this helps somebody

**Comment by GideonWolfe** [comment]
I am experiencing this after I successfully added the mimetype mapping and scanned my files.

Yesterday, clicking my kbdx file would open it in KeepWeb. Today, it attempts to trigger a file download...

**Comment by nub3** [comment]
Yes, I've been experiencing that aswell. However, Clicking in the top-bars KeeWeb Icon, directly opens the file without hassle and let's me edit it. I guess, as long as you've only a single kdbx file, it works, even when the direct click on the kdbx stops

**Comment by GideonWolfe** [comment]
@nub3 really? When I click the Keeweb icon in my top bar it opens the menu where I must choose a file. There is no option to select files from Nextcloud, just my local computer. The only way I can actually access my database is by clicking the file.

**Comment by ithurtswhen-IP** [comment]
Here ist the solution:
Yes, `sudo -u www-data php occ files:scan --all` is mandatory!
So i installed [OCCWEB app](https://apps.nextcloud.com/apps/occweb) 
And run `files:scan --all`

Alo generate the File **/config/mimetypemapping.json**
`{
  "kdbx": ["application/x-kdbx"]
}`

**Comment by SimJoSt** [comment]
@EgoLeX does the problem persist with the newest, just released version? https://github.com/jhass/nextcloud-keeweb/releases  
It changes and automates the way the file extension is registered.

**Comment by thenameiwantedwastaken** [comment]
I have this problem with NC 18.06 Keeweb 0.6.2.

**Comment by maurerle** [comment]
Could you uninstall (e.g. disable and remove) and install NC-Keeweb and check if the issue persists? @thenameiwantedwastaken 
Since NC 18 the correct mimetype configuration should be already in the core package.

**Comment by thenameiwantedwastaken** [comment]
I have disabled and removed KeeWeb, then re-installed it. The behaviour is the same -- clicking on a .kbdx file downloads it.

**Comment by Larsene** [comment]
i've the same problem, but the solution from **ithurtswhen-IP** works perfectly

**Comment by SimJoSt** [comment]
I have had no issues since the fix. @thenameiwantedwastaken @Larsene could you supply a system report via the "issue reporting" section in the admin interface. This would help the maintainers locate the issue.

**Comment by Larsene** [comment]
I've no more issue too

**Comment by SimJoSt** [comment]
@EgoLeX is this true for you as well?

**Comment by arnowelzel** [comment]
At some point, the section about MIME type detected got removed: https://github.com/jhass/nextcloud-keeweb/pull/105

However - we still need a MIME type for *.kdbx to get Keeweb working. Is `application/x-kdbx` now officially part of Nextcloud?

**Comment by SimJoSt** [comment]
@arnowelzel are you saying you are still having issues like the issue opener described? Does clicking a kdbx file not open the Keeweb app for you?

**Comment by arnowelzel** [comment]
@SimJoSt no - I never had these issues. I was just talking about what is needed to avoid that.

JFTR: I am one of the maintainers of this app ;-)

**Comment by arnowelzel** [comment]
@EgoLeX Can you please give a feedback, if this issue is fixed for you now? I can't reproduce this here. Opening kdbx files works fine here.
